types/uint.go

```diff
@@ -161,7 +161,7 @@ func (u *Uint) MarshalTo(data []byte) (n int, err error) {
 	if u.i == nil {
 		u.i = new(big.Int)
 	}
-	if len(u.i.Bytes()) == 0 {
+	if u.i.BitLen() == 0 { // The value 0
```

Comment:
@odeke-em mind adding a test case to cover this line, please?

Reply:
Roger that, I've added some tests for both Int and Uint, please take a look!